server/routes/category.js

```diff
@@ -32,6 +34,21 @@ router.get('/getAll', async (req, res) => {
         res.status(500).json({message: error.message})
     }
 })
+
+// Get by Name Method
+router.get('/getOne/:name', async (req, res) => {
+    try{
```

Comment:
Nirmal will hopefully explain this soon, but API routes should be built in a "ReSTful" manner. You can look up ReSTful APIs yourself but the core idea is that it should be designed around *resources,* and each route's name should reflect what resource it touches.

This route appears to be intended for retrieving a single category. Therefore the route should be named `/category`. The HTTP verb (GET in this case) indicates what the action is (retrieve), so it should not be repeated in the route name. The combination of the two, `GET /category`, tells you that you're retrieving a category.

IIRC you can tell Express to give each of the routes in this router a prefix, which I hope you guys do, so if you prefix all of these category-related routes with `/category` then this particular GET route only needs to be named `/` in this file.

Immediately after this route, there is another one with the exact same name: `GET /getOne/:id`. I'm not sure how the application would differentiate between getting by name and getting by ID. Both of these routes do the same thing (retrieve a single category) in different ways (by name and by ID), so I would combine them into a single `GET /category` and use a query parameter to distinguish how you want to retrieve the category. Then you would have:
```
GET /category?name=
GET /category?id=
```
which is more in line with industry standards.